### PR TITLE
Update upgrade guide info

### LINKED ISSUE #2174: Polymer 2.0 API Upgrade CheatSheet
_From @pkaske on May 20, 2017 18:58_

I started a cheat sheet for reference with _before_ and _after_ examples when upgrading elements to the Polymer 2.0 api. I found this easier than searching the official docs all the time.

Maybe this is useful for somebody else, so I though I open an issue here.  
(feel free to close if this isn't the right place)

The doc is by no means complete. In fact I just copy and pasted the last 5 minutes from the official doc to get this started. PRs with "cheats" you would like to have for easy access very welcome.

**The cheat sheet is only meant for upgrading to class based elements, right now. Not hybrid.**

https://github.com/pkaske/polymer2.0-upgrade-cheatsheet

_Copied from original issue: Polymer/polymer#4615_

### PR DESCRIPTION
**Pull request by katejeffreys**
Fixes #2174

### CODE REVIEW

File: app/2.0/docs/upgrade.md [hunk 1]
```diff
@@ -787,13 +782,47 @@ this._observer = new Polymer.FlattenedNodesObserver(this._nodesChanged);
...
+this.getContentChildren
+```
+
+2.x {.caption}
+
+```js
+this.shadowRoot
+  .querySelector('slot')
+  .assignedNodes({flatten:true})
+  .filter(n.nodeType === Node.ELEMENT_NODE)
```
Comment (pkaske) [inline app/2.0/docs/upgrade.md]: The filter function syntax is wrong.
Should be `.filter(n => n.nodeType === Node.ELEMENT_NODE)`

File: app/2.0/docs/upgrade.md [hunk 2]
```diff
@@ -787,13 +782,47 @@ this._observer = new Polymer.FlattenedNodesObserver(this._nodesChanged);
...
+
+```js
+this.getEffectiveChildren
+```
+
+2.x {.caption}
+
+```js
+let effectiveChildren = 
+  Polymer.FlattenedNodesObserver.getFlattenedNodes(this).filter(n.nodeType === Node.ELEMENT_NODE)
```
Comment (pkaske) [inline app/2.0/docs/upgrade.md]: Should be `.filter(n => n.nodeType === Node.ELEMENT_NODE)`

File: app/2.0/docs/upgrade.md [hunk 3]
```diff
@@ -899,6 +926,24 @@ After {.caption}
 this.updateStyles({'--my-dynamic-property': 'red'});
 ```
 
+To handle cases in which `getComputedStyleValue` was previously used, use the custom ShadyCSS API when the polyfill is loaded:
+
+Before {.caption}
+
+```
+this.getComputedStyleValue()
```
Comment (arthurevans) [inline app/2.0/docs/upgrade.md]: This should be `style = this.getComputedStyleValue('--something');`

File: app/2.0/docs/upgrade.md [hunk 4]
```diff
@@ -787,13 +782,47 @@ this._observer = new Polymer.FlattenedNodesObserver(this._nodesChanged);
 In addition, `Polymer.FlattenedNodesObserver.getFlattenedNodes(node)` can be used to replace the
 `getEffectiveChildNodes` method.
 
+To replace the `getEffectiveChildren` method, use the `getFlattenedNodes` helper method, and filter down to just the elements (ignore comments and text nodes):
+
+1.x {.caption}
+
+```js
+this.getEffectiveChildren
```
Comment (arthurevans) [inline app/2.0/docs/upgrade.md]: `var effectiveChildren = this.getEffectiveChildren();`

File: app/2.0/docs/upgrade.md [hunk 5]
```diff
@@ -787,13 +782,47 @@ this._observer = new Polymer.FlattenedNodesObserver(this._nodesChanged);
...
+let effectiveChildren = 
+  Polymer.FlattenedNodesObserver.getFlattenedNodes(this).filter(n.nodeType === Node.ELEMENT_NODE)
+```
+
+To replace the `getContentChildren` method, write platform code to perform this functionality (get the `assignedNodes`, and filter down to just the elements, ignoring comments and text nodes):
+
+1.x {.caption}
+
+```
+this.getContentChildren
```
Comment (arthurevans) [inline app/2.0/docs/upgrade.md]: `this.getContentChildren();`

(Or add a variable in both this example and the following one, `var contentChildren =` for 1.x, `let contentChildren =` for 2.x.

File: app/2.0/docs/upgrade.md [hunk 6]
```diff
@@ -787,13 +782,47 @@ this._observer = new Polymer.FlattenedNodesObserver(this._nodesChanged);
...
+```
+
+2.x {.caption}
+
+```js
+this.shadowRoot
+  .querySelector('slot')
```
Comment (arthurevans) [inline app/2.0/docs/upgrade.md]: Maybe add a comment above the querySelector here, such as:
```
// If you have more than one slot, you can use a 
// different selector to identify the slot you're interested in.
```

**Comment by katejeffreys** [comment]
@arthurevans Feedback addressed. Is it okay to merge?

**Comment by arthurevans** [comment]
LGTM, thanks!

**Pull request merged**